<launch>
    <arg name="launch_prefix" default="" />
    <arg name="node_namespace" default="apriltag_ros_continuous_node" />
    <arg name="camera_name" default="/usb_cam" />
    <arg name="image_topic" default="image_raw" />

    <!-- Set parameters -->
    <rosparam command="load" file="$(find apriltag_ros)/config/settings.yaml" ns="$(arg node_namespace)" />
    <rosparam command="load" file="$(find apriltag_ros)/config/tags.yaml" ns="$(arg node_namespace)" />

    <!-- stop detector -->
    <node name="stop_detector" pkg="lab4" type="stop_detector.py" output="screen">
        <param name="detector" value="/home/dfec/master_ws/src/ece387_master_sp23-JunhyungPark1821/master/lab4/training_images/stop_detector.svm"/>
    </node>

    <!-- apriltag distance -->
    <node name="apriltag_dist" pkg="lab4" type="apriltag_dist.py" output="screen">
    </node>

    <machine
        name="robot5"
        address="robot5"
        env-loader="/home/pi/robot_ws/devel/remote_env_loader.sh"
        default="true"
        user="pi"
    />

    <!-- usb camera -->
    <node machine="robot5" name="usb_cam" pkg="usb_cam" type="usb_cam_node" output="screen" >
        <param name="video_device" value="/dev/video0" />
        <param name="image_width" value="320" />
        <param name="image_height" value="240" />
        <param name="pixel_format" value="yuyv" />
        <param name="camera_frame_id" value="usb_cam" />
        <param name="io_method" value="mmap"/>
    </node>

    <!-- apriltag_ros -->
    <node machine="robot5" pkg="apriltag_ros" type="apriltag_ros_continuous_node" name="$(arg node_namespace)" clear_params="true" output="screen" launch-prefix="$(arg launch_prefix)" >
        <!-- Remap topics from those used in code to those on the ROS network -->
        <remap from="image_rect" to="$(arg camera_name)/$(arg image_topic)" />
        <remap from="camera_info" to="$(arg camera_name)/camera_info" />

        <param name="publish_tag_detections_image" type="bool" value="true" />      <!-- default: false -->
    </node>
</launch>